x-pack/plugin/sql/qa/src/main/resources/agg.sql-spec

```diff
@@ -467,6 +467,19 @@ SELECT languages, COUNT(*) c FROM test_emp GROUP BY languages HAVING NOT COUNT(*
 selectLangGroupByLangHavingDifferent
 SELECT languages, COUNT(*) c FROM test_emp GROUP BY languages HAVING COUNT(*) <> 1 ORDER BY languages DESC;
 
+aggWithHavingAndComparisonOnNaN1
+SELECT max(languages) AS maxl, gender FROM "test_emp" WHERE emp_no IN(10018, 10019, 10020, 10021) GROUP BY gender HAVING maxl <= 2;
```

Comment:
Can you please try other aggs as well?
For example use min and max to trigger a stats agg, kurtosis for matrix, sum (which should always return zero), percentiles and percentile ranks.
If indeed the bucket selector gives us just the value, I think the proper fix should actually be inside ES...
@polyfractal this looks like another instanceof https://github.com/elastic/elasticsearch/issues/34903 - what do you think?

Reply:
Sorry, playing catch-up here.  What's the underlying aggregation structure being generated?  Aggs + bucket_selector?  Which value(s) being aggregated are NaN/null?

Reply:
In the example above `HAVING maxl <=2` becomes a Painless script doing a bucket selector pointing to `MAX` agg with the content of (`max_agg <=2`). 
This issue appears though for any agg - when running against a null bucket, instead of returning nulls, the aggs selected by the bucket selector their default values (like in #34903). In case of Max is a `NaN` but this clearly differs from case to case.
In scripting however we have no understanding of the source so we can't determine that it's an agg since we only receive the value (`NaN`) or whatever.

I would argue this is an actual bug in ES.

Reply:
@polyfractal ^^

Reply:
@costin Ahh, yes I see.  I agree it's related to #34903.  Bucket selector is just resolving to the internal agg object and asking for it's `value()`, which returns the primitive `double` internal state unlike xcontent which does the null conversion.

I think we can probably expose the `hasValue()` method (or whatever the mechanism ends up being) in the bucket_selector painless context so that it can be used from scripts.  I don't think it's something we can do automatically though, because of how the pipeline framework works right now.  It's a similar issue to #27377 (not quite the same, but similar root issue).  I don't think we can fix that easily until the transport client goes away, or else we'll have a big breaking change to how agg objects work (`double` to `Double` or `Optional`)

So the script might need to do a check if there is a value before actually using the value.  Something like:

```
if (hasValue(max1) && max1.value <= 2) {
   ...
}
```

Not ideal but probably the path of least resistance right now. Or maybe introduce a new `valueOrNull()` method in the context or something.

I'll make a note on #34903 about pipelines